### PR TITLE
Stop using Advertising Id in track and match requests

### PR DESCRIPTION
**Pull request by dimamo5**
Depends on #38 

Fetching and using the Advertising Id implicitly on `track` and `match` requests may lead the client app to break some privacy rules. It is also problematic on devices that don't use Google services or in devices with have stricter privacy settings. To avoid dealing with such issues we decided to shift the responsibility of using Google's Advertising Id to the client app.

So, this PR changes the `track` method to accept an `UserId` and stops fetching the Advertising Id on `init` and using it on track and match requests.

### CODE REVIEW

**Review by silva95gustavo — APPROVED** [review]

File: velocidi-sdk/src/test/kotlin/com/velocidi/VelocidiTest.kt [hunk 1]
```diff
@@ -78,13 +75,16 @@ class VelocidiTest {
 
         val context: Context = ApplicationProvider.getApplicationContext()
 
-        Velocidi.instance = VelocidiMockSync(config, context)
+        Velocidi.instance = Velocidi(config, context)
 
-        Velocidi.getInstance().match("provider1", listOf(UserId("eml", "mail@example.com")))
+        Velocidi.getInstance().match(
+            "provider1",
+            listOf(UserId("123"), UserId("mail@example.com", "eml"))
```
Comment (thyandrecardoso) [inline velocidi-sdk/src/test/kotlin/com/velocidi/VelocidiTest.kt:82]: Nitpick: can we remove literal email examples? This example in particular is not consistent with the id _type_ and I think we should avoid exemplifying plain emails... Can we use a proper sha256 example here and in other examples?

**Review by thyandrecardoso — CHANGES_REQUESTED** [review]

File: velocidi-sdk/src/main/kotlin/com/velocidi/UserId.kt [hunk 2]
```diff
@@ -3,7 +3,10 @@ package com.velocidi
 /**
  * Data model of a user identifier
  *
- * @property type type of identifier (e.g. EML)
+ * @property type type of identifier (defaults to "gaid")
  * @property id the identifier
+ * @see <a href="https://docs.velocidi.com/collect/user-ids">https://docs.velocidi.com/collect/user-ids</a>
  */
-data class UserId(val type: String, val id: String)
+data class UserId(val id: String, val type: String = "gaid") {
```
[the remark this answers is not in the document]
Comment (dimamo5) [reply velocidi-sdk/src/main/kotlin/com/velocidi/UserId.kt:11]: Fixed in f19f929

File: velocidi-sdk/src/main/kotlin/com/velocidi/Velocidi.kt [hunk 3]
```diff
@@ -109,22 +66,27 @@ open class Velocidi internal constructor(val config: Config, context: Context) {
...
-     * Match Google Advertising Id with the list of provided user ids
+     * Identify a user across multiple channels
+     *
+     * For more information, see https://docs.velocidi.com/collect/matches
      *
      * @param providerId Id of the match provider
-     * @param userIds List of user ids to be linked along with Advertising Id
+     * @param userIds List of user ids to be linked
      */
     fun match(providerId: String, userIds: List<UserId>) {
```
[the remark this answers is not in the document]
Comment (dimamo5) [reply velocidi-sdk/src/main/kotlin/com/velocidi/Velocidi.kt:88]: Done in 87c2b67

**Review by thyandrecardoso — APPROVED** [review]

**Comment by thyandrecardoso** [comment]
Looks good. Waiting on the approval from @silva95gustavo.

**Comment by silva95gustavo** [comment]
Good job @dimamo5!

**Pull request merged**